packages/collection/src/collection.ts

```diff
@@ -274,6 +274,60 @@ export class Collection<K, V> extends Map<K, V> {
 		return undefined;
 	}
 
+	/**
+	 * Searches for a last item where the given function returns a truthy value. This behaves like
+	 * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast | Array.findLast()}.
+	 *
+	 * @param fn - The function to test with (should return boolean)
+	 * @param thisArg - Value to use as `this` when executing function
+	 */
+	public findLast<V2 extends V>(fn: (value: V, key: K, collection: this) => value is V2): V2 | undefined;
+	public findLast(fn: (value: V, key: K, collection: this) => unknown): V | undefined;
+	public findLast<This, V2 extends V>(
+		fn: (this: This, value: V, key: K, collection: this) => value is V2,
+		thisArg: This,
+	): V2 | undefined;
+	public findLast<This>(fn: (this: This, value: V, key: K, collection: this) => unknown, thisArg: This): V | undefined;
+	public findLast(fn: (value: V, key: K, collection: this) => unknown, thisArg?: unknown): V | undefined {
+		if (typeof fn !== 'function') throw new TypeError(`${fn} is not a function`);
+		if (thisArg !== undefined) fn = fn.bind(thisArg);
+		const entries = [...this.entries()].reverse();
+		for (const [key, val] of entries) {
```

Comment:
We can do an inverse loop (`this.size - 1..0`) instead of creating another array.

Reply:
yeah my original idea, but then typescript there's that it is possibly undefined, yeah forcing it is an option
and the way looks cleaner
but it can be done as it's faster

Reply:
Also saying to not make another array is a little bit wrong
as still there must be an array with entires